intel/AiCSD
# feat: add grpc-go pipeline service to work with the pipeline validator

## PR Checklist

## Author Mandatory (to be filled by PR Author/Submitter)

### CODE MAINTAINABILITY

    - <https://github.com/intel/AiCSD/blob/main/.github/CONTRIBUTING.md>

### What are the steps to test this PR contribution?

How does a reviewer know it works?

1. add the Aicsd code as a submodule under retail-use-cases/use-cases
2. change directories to retail-use-cases
3. download models and sample media `make download-models download-sample-media`
4. build the ovms server `make build-ovms-server`
5. change directories to use-cases/aicsd
6. build the pipeline validator and pipeline grpc-go services `make docker-pipeline-val docker-pipeline-grpc-go`
7. modify docker-compose-pipeline-val.yml 
    ```
    APPLICATIONSETTINGS_PIPELINEHOST: pipeline-grpc-go
    APPLICATIONSETTINGS_PIPELINEPORT: 59790
    ```
8. run the pipeline validator and pipeline grpc-go services `make run-pipeline-grpc-go`
9. send a POST request to `http://localhost:59788/api/v1/launchPipeline` with the body
    ```
    {
    "InputFileLocation":"rtsp://camera-simulator:8554/camera_0",
    "PipelineTopic":"ovms-grpc/yolov5",
    "OutputFileFolder":"0.0.0.0:8555",
    "ModelParams":
      {
        "model_name":"yolo",
        "model_version":"v5"
      }
    }
    ```
10. Check the log of the pipeline-grpc-go container to see that there are averages and that the pipeline is running.
11. To allow the pipeline to continue, stop the camera simulator containers to stop the media stream. This will allow the pipeline-grpc-go pipeline to complete.
12. Check the status of the job to see that the pipeline is complete by making a get request to `http://localhost:59788/api/v1/job`
13. Restart the camera simulator containers if desired to run the launch command again.

To clean up, run 'make down clean-files clean-volumes`

Are there any special instructions beyond standard workflow?

this is a single system service.

## Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)

### QUALITY CHECKS


## Other information
